### PR TITLE
cmake cleanup, docs for switches, config output

### PR DESCRIPTION
**Pull request by mpreisler**
cmake in openscap master had pretty bad docs.

Exhibit 1:
```
# CURL
find_package(CURL)

# DBUS
find_package(DBus)

# DOXYGEN
find_package(Doxygen)
```

Exhibit 2:
```
# ENABLES PYTHON3
set(ENABLE_PYTHON3 FALSE CACHE BOOL "enables Python3")

# ENABLES PERL
set(ENABLE_PERL FALSE CACHE BOOL "enables Perl")

# ENABLES VALGRIND
set(ENABLE_VALGRIND FALSE CACHE BOOL "enables Valgrind")

# ENABLES SSP
set(ENABLE_SSP FALSE CACHE BOOL "enables fstack-protector")

# ENABLES SCE
set(ENABLE_SCE FALSE CACHE BOOL "enables SCE")

# ENABLES PROBES
set(ENABLE_PROBES TRUE CACHE BOOL "enables probes")
```

After change:
```
$ cmake .
-- OpenSCAP 1.3.0
-- (see /home/mpreisle/d/openscap/README.md for build instructions)
--  
-- Checking for module 'acl'
--   Package 'acl', required by 'virtual:world', not found
-- Found ACL 
-- Found BLKID 
-- Found GLib 
-- Found GObject 
-- Found GCONF 
-- Found OPENDBX 
-- Found POPT 
-- Found PROCPS 
-- Found RPM 4.13.0.2
-- Found SELINUX 
--  
-- CMake:
-- generator: Unix Makefiles
-- source directory: /home/mpreisle/d/openscap
-- build directory: /home/mpreisle/d/openscap/build
--  
-- Core features:
-- CCE: OFF
-- SCE: OFF
--  
-- OVAL:
-- base probe support: ON
-- SEAP msgid bit-size: 32
-- independent probes: ON
-- unix probes: ON
-- linux probes: ON
-- solaris probes: OFF
--  
-- Language bindings:
-- python2 bindings: ON
-- python3 bindings: OFF
-- perl bindings: OFF
```

```
# ---------- CORE FEATURE SWITCHES

option(ENABLE_CCE "enables support for Common Configuration Enumeration" FALSE)
option(ENABLE_SCE "enables Script Check Engine - an alternative checking engine that let's you use executables instead of OVAL for checks" FALSE)

# ---------- OVAL FEATURE SWITCHES

option(ENABLE_PROBES "build OVAL probes - each probe implements an OVAL test" TRUE)
set(SEAP_MSGID_BITS 32 CACHE STRING "Size of SEAP_msgid_t in bits [32|64]")
option(ENABLE_PROBES_INDEPENDENT "build OVAL probes for independent (cross platform) OVAL tests" TRUE)
# TODO: We probably only want these on UNIX systems
option(ENABLE_PROBES_UNIX "build OVAL probes for the UNIX OVAL tests" TRUE)
string(COMPARE EQUAL ${CMAKE_SYSTEM_NAME} "Linux" IS_LINUX)
option(ENABLE_PROBES_LINUX "build OVAL probes for the Linux OVAL tests" ${IS_LINUX})
string(COMPARE EQUAL ${CMAKE_SYSTEM_NAME} "Solaris" IS_SOLARIS)
option(ENABLE_PROBES_SOLARIS "build OVAL probes for the Solaris OVAL tests" ${IS_SOLARIS})

# ---------- EXECUTABLES / UTILITIES SWITCHES

option(ENABLE_OSCAP_UTIL "build the oscap utility, oscap is the core utility for evaluation and processing of SCAP data" TRUE)
option(ENABLE_OSCAP_UTIL_AS_RPM "enable the scap-as-rpm utility, this let's you package SCAP data as RPMs" TRUE)
```

### CODE REVIEW

**Comment by mpreisler** [comment]
I also think the switches should have a prefix so that you can group them in cmake-gui. Similar to what we do in SSG. If we decide to do that it should be in a different PR.

**Comment by mpreisler** [comment]
I did screw something up with this, right now I don't see what it is, any feedback appreciated. Will look more into this tomorrow.

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,22 @@ set(OPENSCAP_VERSION_PATCH "0")
 set(OPENSCAP_VERSION "${OPENSCAP_VERSION_MAJOR}.${OPENSCAP_VERSION_MINOR}.${OPENSCAP_VERSION_PATCH}")
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
 
+message(STATUS "OpenSCAP ${OPENSCAP_VERSION}")
+message(STATUS "(see ${CMAKE_SOURCE_DIR}/README.md for build instructions)")
+message(STATUS " ")
+
+# Strictly speaking in-source will work but will be very messy, let's
+# discourage our users from using them
+if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
```
Comment (matejak) [inline CMakeLists.txt]: As a user, I would prefer to see the errors right away, not having to scroll up in order to see why the configuration has failed. Right now, the configuration is very fast, and this error message related to it is neither at the beginning nor at the bottom of the output.
There are also too many newlines in the error message that is print to the screen, but it seems that nothing can be done about that.

**Comment by matejak** [comment]
I like the transition from `set` directives to `option`.

File: CMakeLists.txt (hunk 1 above)
Comment (mpreisler) [reply CMakeLists.txt]: I straight up stole this from SSG :-) I have made a PR there, check it out please. If it's accepted I can do the same here.

https://github.com/OpenSCAP/scap-security-guide/pull/2540

Comment (mpreisler) [reply CMakeLists.txt]: The SSG PR was accepted so I have done the same here as well. It fatal errors now.

**Comment by mpreisler** [comment]
> However, options are either ON or OFF, not TRUE or FALSE. Therefore, the conditional in tests/test-common.sh.in:23 has to be updated.

Great catch. I will fix that.

File: CMakeLists.txt (hunk 1 above)
Comment (matejak) [reply CMakeLists.txt]: I have tried that one, it works nicely.

**Comment by jan-cerny** [comment]
@mpreisler I really like this improvement. It is great!

**Comment by mpreisler** [comment]
Failed in rpmverify, unrealted to this PR.

@openscap-ci test this please

**Comment by mpreisler** [comment]
@openscap-ci test this please

**Comment by mpreisler** [comment]
@openscap-ci test this please

**Comment by mpreisler** [comment]
I hate you, Jenkins...

@openscap-ci test this please

**Comment by mpreisler** [comment]
@openscap-ci test this please

**Comment by mpreisler** [comment]
@openscap-ci test this please

**Comment by mpreisler** [comment]
@openscap-ci test this please

**Comment by mpreisler** [comment]
@openscap-ci test this please

**Comment by matejak** [comment]
This is a great step forward in terms of features and maintainability, many thanks!

**Pull request merged**